appengine/pond_storage/opponents.py

```diff
@@ -60,7 +60,8 @@ def get_entries_in_range(user_entry, opponent_entries, min_rank, max_rank):
       LeaderboardEntry.ranking <= max_rank
   ).fetch(_FETCH_LIMIT)
   new_entries = [e for e in entries
-                 if e not in opponent_entries and e.key != user_entry.key]
+                 if e not in opponent_entries and e.key != user_entry.key and
+                 e.has_duck == True]
```

Comment:
Instead of filtering afterward, you can add to query. rank.py L123 has example. It can be added because its an equality check.